### PR TITLE
[FIX] Fix channels and groups addAll users to a group/channel, when channel is readOnly

### LINKED ISSUE #12058: [BUG] Broadcast Channel & /v1/api/channels.addAll
### Description:

If I create a broadcast channel - all users should be muted or read only in some way. If I test this manually this works perfect. (Adding a user from the WebGUI)

I tried to use the Rest-API for this with a broadcast channel - the api seems to be not evaluating if that is a broadcast channel, results in a writable channel for all users added via the rest-api. (/v1/api/channels.addAll)

### Steps to reproduce:

1. Create a Broadcast Channel
2. Add all users from the server to this channel via the api /v1/api/channels.addAll
3. All Users added can read / write on a write-protected channel

### Expected behavior:

All users added should only have read permissions like it works from the webgui.

### Actual behavior:

All users can write in a writeprotected channel.

image

image

image

You can see that this user has no explicit permission to write to this channel, but can write in the channel like a content-moderator or administrator.

### Server Setup Information:

- Version of Rocket.Chat Server: 0.69.2
- Operating System: Ubuntu Server 16.04 LTS
- Deployment Method: docker
- Number of Running Instances: 3
- DB Replicaset Oplog: yes
- NodeJS Version: v8.11.3
- MongoDB Version: 3.6

### Additional context
Adding something later here.

### PR DESCRIPTION
**Pull request by MarcosSpessatto**
Closes #12058

### CODE REVIEW

File: server/methods/addAllUserToRoom.js [hunk 1]
```diff
@@ -43,6 +43,12 @@ Meteor.methods({
 				RocketChat.models.Messages.createUserJoinWithRoomIdAndUser(rid, user, {
 					ts: now,
 				});
+				if (room.ro) {
+					Meteor.call('muteUserInRoom', {
```
Comment (sampaiodiego) [inline server/methods/addAllUserToRoom.js]: this is rather an expensive method to call for each user and will flood the channel with "user muted" messages.

I'd recommend to change this call to an update on room's object adding all users at the same time to the `muted` field.

**Comment by sampaiodiego** [comment]
@Hudell how does this conflicts with the improvements you were doing on read only channels?

**Comment by Hudell** [comment]
I'll need to merge them manually, as some things done here have changed there.

**Comment by MarcosSpessatto** [comment]
@sampaiodiego @Hudell any updates?

**Pull request closed without merging**